src/javascript/util.ts

```diff
@@ -217,3 +219,123 @@ export function tryResolveDependencyVersion(
   }
   return manifest?.version;
 }
+
+/**
+ * Whether the given dependency version is installed
+ *
+ * This can be used to test for the presence of certain versions of devDependencies,
+ * and do something dependency-specific in certain Components. For example, test for
+ * a version of Jest and generate different configs based on the Jest version.
+ *
+ * NOTE: The implementation of this function currently is currently
+ * approximate: to do it correctly, we would need a separate implementation
+ * for every package manager, to query its installed version (either that, or we
+ * would code to query `package-lock.json`, `yarn.lock`, etc...).
+ *
+ * Instead, we will look at `package.json`, and assume that the versions
+ * picked by the package manager match ~that. This will work well enough for
+ * major version checks, but may fail for point versions.
+ *
+ * What we SHOULD do is: `actualVersion ∈ checkRange`.
+ *
+ * What we do instead is a slightly more sophisticated version of
+ * `requestedRange ∩ checkRange != ∅`. This will always give a correct result if
+ * `requestedRange ⊆ checkRange`, but may give false positives when `checkRange
+ * ⊆ requestedRange`.
+ *
+ * May return `undefined` if the question cannot be answered (for example, if
+ * the dependency is requested via local file dependencies).
+ *
+ * This API may eventually be added to the public projen API, but only after
+ * we implement exact version checking.
+ *
+ * @param dependencyName The name of the dependency
+ * @param checkRange A particular version, or range of versions.
+ */
+export function hasDependencyVersion(
+  project: Project,
+  dependencyName: string,
+  checkRange: string
+): boolean | undefined {
+  const file = NodePackage.of(project)?.file;
+  if (!file) {
+    throw new Error(`Project does not have a NodePackage component`);
```

Comment:
This seems like the same situation as the next one. Let's not treat them differently.

```suggestion
    return undefined; 
```